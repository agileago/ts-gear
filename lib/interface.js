"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const types_1 = require("./tool/types");
exports.httpMethods = types_1.tuple('get', 'put', 'post', 'delete', 'options', 'head', 'patch');
