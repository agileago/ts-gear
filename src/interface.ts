import { JSONSchema4 } from 'json-schema'

export type JSONSchema = JSONSchema4

/** json schema每个节点traverse结构 */
export interface ISchemaNode {
  value: any
  key: string
  parent: any
  path: string[]
}

/** 每个请求参数的原始schema结构 */
export interface IParameter {
  name: string
  in: keyof IParameterSchema
  description: string
  required: boolean
  type: string
  items?: JSONSchema
  format?: string
  schema?: JSONSchema
}

// interface IResponseSchema extends JSONSchema {
//   title?: string
// }

interface IResponse {
  // '200'?: {
  //   description: string
  //   schema: IResponseSchema
  // }
  [key: string]: {
    description: string
    schema?: JSONSchema
  }
}

/** swagger paths内每个http请求详情对应的结构 */
export interface IRequestDetail {
  parameters?: IParameter[]
  summary?: string
  description?: string
  operationId?: string
  produces: string[]
  tags: string[]
  responses: IResponse
  security?: any
  deprecated: boolean
}

interface IRequest {
  /** http method */
  [path: string]: IRequestDetail
}

/** swagger "paths" */
export interface IPaths {
  /** api url path */
  [k: string]: IRequest
}

/** 每个请求的json schema定义的parameters的住装数据结构 */
export interface IParameterSchema {
  query?: JSONSchema
  body?: JSONSchema
  path?: JSONSchema
  formData?: JSONSchema
  header?: JSONSchema
}

export type TPathMatcherFunction = RegExp | ((url: string) => boolean)

export interface IProject {
  /** 项目名字，会在dest文件夹中创建该项目名称的目录 */
  name: string
  /** 读取swagger配置的路径，可以是本地json文件或远程swagger地址
   * http开头的为远程地址
   * 否则按本地文件读取
   * */
  source: string
  /** 需要验证参数在这里添加 */
  fetchOption?: RequestInit
  /** 过滤需要的请求路径
   * 有些项目其中掺杂了很多无用的api，过滤掉可减少文件体积，加速运行
   * */
  pathMatcher?: TPathMatcherFunction
}

/** 用户配置文件定义 */
export interface IUserConfig {
  /** 项目输出文件夹，会在该文件夹下建立以每个项目名建立独立的文件夹
   * 默认为 './service'
   * */
  dest: string

  projects: IProject[]
}
