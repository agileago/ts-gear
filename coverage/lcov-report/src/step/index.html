
<!doctype html>
<html lang="en">

<head>
    <title>Code coverage report for src/step</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../prettify.css" />
    <link rel="stylesheet" href="../../base.css" />
    <link rel="shortcut icon" type="image/x-icon" href="../../favicon.png" />
    <meta name="viewport" content="width=device-width, initial-scale=1" />
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../sort-arrow-sprite.png);
        }
    </style>
</head>
    
<body>
<div class='wrapper'>
    <div class='pad1'>
        <h1><a href="../../index.html">All files</a> src/step</h1>
        <div class='clearfix'>
            
            <div class='fl pad1y space-right2'>
                <span class="strong">67.24% </span>
                <span class="quiet">Statements</span>
                <span class='fraction'>195/290</span>
            </div>
        
            
            <div class='fl pad1y space-right2'>
                <span class="strong">49.17% </span>
                <span class="quiet">Branches</span>
                <span class='fraction'>59/120</span>
            </div>
        
            
            <div class='fl pad1y space-right2'>
                <span class="strong">62.07% </span>
                <span class="quiet">Functions</span>
                <span class='fraction'>36/58</span>
            </div>
        
            
            <div class='fl pad1y space-right2'>
                <span class="strong">67.02% </span>
                <span class="quiet">Lines</span>
                <span class='fraction'>191/285</span>
            </div>
        
            
        </div>
        <p class="quiet">
            Press <em>n</em> or <em>j</em> to go to the next uncovered block, <em>b</em>, <em>p</em> or <em>k</em> for the previous block.
        </p>
    </div>
    <div class='status-line medium'></div>
    <div class="pad1">
<table class="coverage-summary">
<thead>
<tr>
   <th data-col="file" data-fmt="html" data-html="true" class="file">File</th>
   <th data-col="pic" data-type="number" data-fmt="html" data-html="true" class="pic"></th>
   <th data-col="statements" data-type="number" data-fmt="pct" class="pct">Statements</th>
   <th data-col="statements_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="branches" data-type="number" data-fmt="pct" class="pct">Branches</th>
   <th data-col="branches_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="functions" data-type="number" data-fmt="pct" class="pct">Functions</th>
   <th data-col="functions_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="lines" data-type="number" data-fmt="pct" class="pct">Lines</th>
   <th data-col="lines_raw" data-type="number" data-fmt="html" class="abs"></th>
</tr>
</thead>
<tbody><tr>
	<td class="file high" data-value="assembleSchemaToGlobal.ts"><a href="assembleSchemaToGlobal.ts.html">assembleSchemaToGlobal.ts</a></td>
	<td data-value="96.55" class="pic high">
	<div class="chart"><div class="cover-fill" style="width: 96%"></div><div class="cover-empty" style="width: 4%"></div></div>
	</td>
	<td data-value="96.55" class="pct high">96.55%</td>
	<td data-value="29" class="abs high">28/29</td>
	<td data-value="91.67" class="pct high">91.67%</td>
	<td data-value="12" class="abs high">11/12</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="6" class="abs high">6/6</td>
	<td data-value="96.43" class="pct high">96.43%</td>
	<td data-value="28" class="abs high">27/28</td>
	</tr>

<tr>
	<td class="file medium" data-value="cleanRefAndDefinitionName.ts"><a href="cleanRefAndDefinitionName.ts.html">cleanRefAndDefinitionName.ts</a></td>
	<td data-value="78.57" class="pic medium">
	<div class="chart"><div class="cover-fill" style="width: 78%"></div><div class="cover-empty" style="width: 22%"></div></div>
	</td>
	<td data-value="78.57" class="pct medium">78.57%</td>
	<td data-value="14" class="abs medium">11/14</td>
	<td data-value="83.33" class="pct high">83.33%</td>
	<td data-value="6" class="abs high">5/6</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="3" class="abs high">3/3</td>
	<td data-value="78.57" class="pct medium">78.57%</td>
	<td data-value="14" class="abs medium">11/14</td>
	</tr>

<tr>
	<td class="file high" data-value="colletRefsInRequestAndPatchDefinition.ts"><a href="colletRefsInRequestAndPatchDefinition.ts.html">colletRefsInRequestAndPatchDefinition.ts</a></td>
	<td data-value="100" class="pic high">
	<div class="chart"><div class="cover-fill cover-full" style="width: 100%"></div><div class="cover-empty" style="width: 0%"></div></div>
	</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="23" class="abs high">23/23</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="6" class="abs high">6/6</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="7" class="abs high">7/7</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="23" class="abs high">23/23</td>
	</tr>

<tr>
	<td class="file high" data-value="fetchSwagger.ts"><a href="fetchSwagger.ts.html">fetchSwagger.ts</a></td>
	<td data-value="100" class="pic high">
	<div class="chart"><div class="cover-fill cover-full" style="width: 100%"></div><div class="cover-empty" style="width: 0%"></div></div>
	</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="19" class="abs high">19/19</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="4" class="abs high">4/4</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="1" class="abs high">1/1</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="19" class="abs high">19/19</td>
	</tr>

<tr>
	<td class="file low" data-value="generateDefinitionContent.ts"><a href="generateDefinitionContent.ts.html">generateDefinitionContent.ts</a></td>
	<td data-value="19.35" class="pic low">
	<div class="chart"><div class="cover-fill" style="width: 19%"></div><div class="cover-empty" style="width: 81%"></div></div>
	</td>
	<td data-value="19.35" class="pct low">19.35%</td>
	<td data-value="31" class="abs low">6/31</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="26" class="abs low">0/26</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="4" class="abs low">0/4</td>
	<td data-value="19.35" class="pct low">19.35%</td>
	<td data-value="31" class="abs low">6/31</td>
	</tr>

<tr>
	<td class="file low" data-value="importAllDefinition.ts"><a href="importAllDefinition.ts.html">importAllDefinition.ts</a></td>
	<td data-value="22.22" class="pic low">
	<div class="chart"><div class="cover-fill" style="width: 22%"></div><div class="cover-empty" style="width: 78%"></div></div>
	</td>
	<td data-value="22.22" class="pct low">22.22%</td>
	<td data-value="9" class="abs low">2/9</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="0" class="abs high">0/0</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="3" class="abs low">0/3</td>
	<td data-value="22.22" class="pct low">22.22%</td>
	<td data-value="9" class="abs low">2/9</td>
	</tr>

<tr>
	<td class="file high" data-value="index.ts"><a href="index.ts.html">index.ts</a></td>
	<td data-value="100" class="pic high">
	<div class="chart"><div class="cover-fill cover-full" style="width: 100%"></div><div class="cover-empty" style="width: 0%"></div></div>
	</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="11" class="abs high">11/11</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="0" class="abs high">0/0</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="0" class="abs high">0/0</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="11" class="abs high">11/11</td>
	</tr>

<tr>
	<td class="file high" data-value="parseGenericType.ts"><a href="parseGenericType.ts.html">parseGenericType.ts</a></td>
	<td data-value="98.63" class="pic high">
	<div class="chart"><div class="cover-fill" style="width: 98%"></div><div class="cover-empty" style="width: 2%"></div></div>
	</td>
	<td data-value="98.63" class="pct high">98.63%</td>
	<td data-value="73" class="abs high">72/73</td>
	<td data-value="86.84" class="pct high">86.84%</td>
	<td data-value="38" class="abs high">33/38</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="19" class="abs high">19/19</td>
	<td data-value="98.57" class="pct high">98.57%</td>
	<td data-value="70" class="abs high">69/70</td>
	</tr>

<tr>
	<td class="file medium" data-value="prepareProjectDirectory.ts"><a href="prepareProjectDirectory.ts.html">prepareProjectDirectory.ts</a></td>
	<td data-value="50" class="pic medium">
	<div class="chart"><div class="cover-fill" style="width: 50%"></div><div class="cover-empty" style="width: 50%"></div></div>
	</td>
	<td data-value="50" class="pct medium">50%</td>
	<td data-value="8" class="abs medium">4/8</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="2" class="abs low">0/2</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="1" class="abs low">0/1</td>
	<td data-value="50" class="pct medium">50%</td>
	<td data-value="8" class="abs medium">4/8</td>
	</tr>

<tr>
	<td class="file low" data-value="translateSchema.ts"><a href="translateSchema.ts.html">translateSchema.ts</a></td>
	<td data-value="20" class="pic low">
	<div class="chart"><div class="cover-fill" style="width: 20%"></div><div class="cover-empty" style="width: 80%"></div></div>
	</td>
	<td data-value="20" class="pct low">20%</td>
	<td data-value="45" class="abs low">9/45</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="24" class="abs low">0/24</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="10" class="abs low">0/10</td>
	<td data-value="20.45" class="pct low">20.45%</td>
	<td data-value="44" class="abs low">9/44</td>
	</tr>

<tr>
	<td class="file low" data-value="writeProject.ts"><a href="writeProject.ts.html">writeProject.ts</a></td>
	<td data-value="35.71" class="pic low">
	<div class="chart"><div class="cover-fill" style="width: 35%"></div><div class="cover-empty" style="width: 65%"></div></div>
	</td>
	<td data-value="35.71" class="pct low">35.71%</td>
	<td data-value="28" class="abs low">10/28</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="2" class="abs low">0/2</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="4" class="abs low">0/4</td>
	<td data-value="35.71" class="pct low">35.71%</td>
	<td data-value="28" class="abs low">10/28</td>
	</tr>

</tbody>
</table>
</div>
                <div class='push'></div><!-- for sticky footer -->
            </div><!-- /wrapper -->
            <div class='footer quiet pad2 space-top1 center small'>
                Code coverage generated by
                <a href="https://istanbul.js.org/" target="_blank">istanbul</a>
                at Mon Apr 27 2020 18:43:36 GMT+0800 (GMT+08:00)
            </div>
        </div>
        <script src="../../prettify.js"></script>
        <script>
            window.onload = function () {
                prettyPrint();
            };
        </script>
        <script src="../../sorter.js"></script>
        <script src="../../block-navigation.js"></script>
    </body>
</html>
    