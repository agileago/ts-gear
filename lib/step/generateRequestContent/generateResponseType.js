"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const lodash_1 = require("lodash");
const source_1 = require("../../source");
const schemaToTypescript_1 = require("../../tool/schemaToTypescript");
const assembleDoc_1 = require("../../tool/assembleDoc");
/**
 * when has responses spec, get an interface type and use the first 2xx member as successType
 * when has not responses, use any as successType
 * */
exports.generateResponseType = (functionName, responses) => {
    const responseTypeName = `I${lodash_1.upperFirst(functionName)}Response`;
    // use first 2xx response type as success response type
    let successTypeContent = `export type ${responseTypeName}Success = any`;
    let responseTypeContent = `export type ${responseTypeName} = any`;
    const successTypeName = `${responseTypeName}Success`;
    const responseStatuses = Object.getOwnPropertyNames(responses).sort();
    if (responseStatuses.length > 0) {
        const source = source_1.sow();
        const inter = source.addInterface({
            name: responseTypeName,
            isExported: true,
        });
        responseStatuses.forEach(status => {
            inter.addProperty({
                name: String(status),
                type: schemaToTypescript_1.schemaToTypescript(responses[status]),
                docs: assembleDoc_1.assembleDoc(responses[status]),
            });
        });
        responseTypeContent = source_1.harvest(source);
        const firstResponseStatus = responseStatuses[0];
        if (firstResponseStatus.startsWith('2') || firstResponseStatus === 'default') {
            if (Number.isNaN(Number(firstResponseStatus))) {
                successTypeContent = `export type ${responseTypeName}Success = PropertyType<${responseTypeName}, '${firstResponseStatus}'>`;
            }
            else {
                successTypeContent = `export type ${responseTypeName}Success = PropertyType<${responseTypeName}, ${firstResponseStatus}>`;
            }
        }
    }
    return {
        responseTypeContent,
        successTypeContent,
        responseTypeName,
        successTypeName,
    };
};
