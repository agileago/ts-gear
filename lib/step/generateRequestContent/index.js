"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.generateRequestContent = void 0;
const os_1 = require("os");
const source_1 = require("../../source");
const transformSwaggerPathToRouterPath_1 = require("../../tool/transformSwaggerPathToRouterPath");
const projectGlobalVariable_1 = require("../../projectGlobalVariable");
const assembleDoc_1 = require("../../tool/assembleDoc");
const constant_1 = require("../../constant");
const generateMockData_1 = require("./generateMockData");
const generateResponseType_1 = require("./generateResponseType");
const generateParameterType_1 = require("./generateParameterType");
const join = require("url-join");
/** from swagger spec paths assemble request functions */
exports.generateRequestContent = (spec, project) => {
    const { pathMatcher, withBasePath, withHost } = project;
    const { requestMap, definitionMap, enumMap } = projectGlobalVariable_1.getGlobal(project);
    const shouldMockResponseStatement = project.shouldMockResponseStatement || constant_1.defaultShouldMockResponseStatement;
    const resultContent = [];
    Object.getOwnPropertyNames(requestMap).forEach((requestFunctionName) => {
        const requestTypeScriptContent = [];
        const request = requestMap[requestFunctionName];
        const { httpMethod } = request;
        if (pathMatcher) {
            if (typeof pathMatcher === 'function') {
                if (!pathMatcher(request.pathName, httpMethod)) {
                    return;
                }
            }
            else if (!pathMatcher.test(request.pathName)) {
                return;
            }
        }
        let parameterTypeName = '';
        let parameterRequired = false;
        if (request.parameters && request.parameters.length > 0) {
            const parameterType = generateParameterType_1.generateParameterType(requestFunctionName, request.parameters);
            parameterTypeName = parameterType.parameterTypeName;
            parameterRequired = parameterType.parameterRequired;
            requestTypeScriptContent.push(parameterType.parameterTypeContent);
        }
        const responseType = generateResponseType_1.generateResponseType(requestFunctionName, request.responses);
        requestTypeScriptContent.push(responseType.responseTypeContent);
        requestTypeScriptContent.push(responseType.successTypeContent);
        const urlPath = join(spec.basePath || '', transformSwaggerPathToRouterPath_1.transformSwaggerPathToRouterPath(String(request.pathName)));
        const source = source_1.sow();
        const requesterStatment = `return requester('${urlPath}', {${withHost ? `, host: '${spec.host}'` : ''}${withBasePath ? `, basePath: '${spec.basePath}'` : ''}method: '${httpMethod}'${parameterTypeName ? ', ...option' : ''}}) as Promise<any>`;
        const functionStatment = `if (${shouldMockResponseStatement}) {
      return Promise.resolve(${requestFunctionName}.mockData as any)
    }
    ${requesterStatment}`;
        const functionData = {
            name: requestFunctionName,
            isExported: true,
            returnType: `Promise<${responseType.successTypeName}>`,
            statements: functionStatment,
            docs: assembleDoc_1.assembleDoc(request.schema),
        };
        functionData.parameters = [];
        if (parameterTypeName) {
            functionData.parameters.push({
                hasQuestionToken: !parameterRequired,
                name: 'option',
                type: parameterTypeName,
            });
        }
        source.addFunction(functionData);
        const mockStatment = `${requestFunctionName}.mockData = (${JSON.stringify(generateMockData_1.generateMockData(request, definitionMap, enumMap))} as any)`;
        source.addStatements(`
if (${shouldMockResponseStatement}) {
  ${mockStatment}
}
${requestFunctionName}.method = '${httpMethod}'
`);
        requestTypeScriptContent.push(source_1.harvest(source));
        /** store typescript content to requestMap */
        request.typescriptContent = requestTypeScriptContent.join(os_1.EOL);
        resultContent.push(request.typescriptContent);
    });
    /** return value only for test and debug */
    return resultContent.join(os_1.EOL);
};
