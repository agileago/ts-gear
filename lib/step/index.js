"use strict";
function __export(m) {
    for (var p in m) if (!exports.hasOwnProperty(p)) exports[p] = m[p];
}
Object.defineProperty(exports, "__esModule", { value: true });
__export(require("./getUserConfig"));
__export(require("./fetchSwagger"));
__export(require("./prepareProjectDirectory"));
__export(require("./translateSchema"));
__export(require("./assembleSchemaToGlobal"));
__export(require("./cleanRefAndDefinitionName"));
__export(require("./parseGenericType"));
__export(require("./colletRefsInRequestAndPatchDefinition"));
__export(require("./generateDefinitionContent"));
__export(require("./generateRequestContent"));
__export(require("./writeProject"));
