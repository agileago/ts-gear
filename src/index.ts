export { requester as fetchRequester } from './requester/fetch'
export { requester as axiosRequester } from './requester/axios'
export { IProject, PropertyType } from './interface'
