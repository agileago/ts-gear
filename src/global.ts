import { IProjectGlobalMap, IProject, HttpMethod } from 'src/interface'
import { clearObject } from 'src/tool/clearObject'

const projectGlobal: IProjectGlobalMap = {}

export const getGlobal = (project: IProject) => {
  if (!projectGlobal[project.name]) {
    projectGlobal[project.name] = {
      definitionMap: {},
      requestMap: {},
      requestRefSet: new Set(),
    }
  }
  return projectGlobal[project.name]
}

export const restore = (project: IProject) => {
  const g = projectGlobal[project.name]
  clearObject(g.definitionMap)
  clearObject(g.requestMap)
  g.requestRefSet.clear()
}

export const httpMethods: HttpMethod[] = ['get', 'put', 'post', 'delete', 'options', 'head', 'patch']
