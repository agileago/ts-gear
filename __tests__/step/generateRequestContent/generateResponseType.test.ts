import { Spec } from 'swagger-schema-official'
import { cloneDeep } from 'lodash'

import { generateResponseType } from 'src/step/generateRequestContent/generateResponseType'
import petSpec from 'example/fixture/pet.json'
import * as step from 'src/step'
import { restore } from 'src/global'
import { IProject } from 'src/interface'

describe('response type content', () => {
  const project: IProject = {
    name: 'pet',
    dest: './service',
    source: 'fixture/pet.json',
    requester: () => Promise.resolve(),
  }
  it('response', () => {
    const schema = cloneDeep(petSpec) as Spec
    step.cleanRefAndDefinitionName(schema, true)
    step.assembleSchemaToGlobal(schema, project)
    let content = generateResponseType('IGetXXX', schema.paths['/pet'].post!.responses)
    console.log(content)
    content = generateResponseType('IGetXXX', schema.paths['/pet'].put!.responses)
    console.log(content)

    content = generateResponseType('IGetXXX', schema.paths['/store/order'].post!.responses)
    console.log(content)
    restore(project)
  })
})
