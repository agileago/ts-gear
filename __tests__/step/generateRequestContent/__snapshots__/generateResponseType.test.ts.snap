// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`response type content response 1`] = `
Object {
  "responseTypeContent": "/** @description response type for GetXXX */
export interface GetXXXResponse {
    /**
     * @description
     *   Invalid input
     */
    405: any;
}
",
  "responseTypeName": "GetXXXResponse",
  "successTypeContent": "export type GetXXXResponseSuccess = any",
  "successTypeName": "GetXXXResponseSuccess",
}
`;

exports[`response type content response 2`] = `
Object {
  "responseTypeContent": "/** @description response type for GetXXX */
export interface GetXXXResponse {
    /**
     * @description
     *   Invalid ID supplied
     */
    400: any;
    /**
     * @description
     *   Pet not found
     */
    404: any;
    /**
     * @description
     *   Validation exception
     */
    405: any;
}
",
  "responseTypeName": "GetXXXResponse",
  "successTypeContent": "export type GetXXXResponseSuccess = any",
  "successTypeName": "GetXXXResponseSuccess",
}
`;

exports[`response type content response 3`] = `
Object {
  "responseTypeContent": "/** @description response type for GetXXX */
export interface GetXXXResponse {
    /**
     * @description
     *   successful operation
     */
    200: Order;
    /**
     * @description
     *   Invalid Order
     */
    400: any;
}
",
  "responseTypeName": "GetXXXResponse",
  "successTypeContent": "export type GetXXXResponseSuccess = GetXXXResponse[200]",
  "successTypeName": "GetXXXResponseSuccess",
}
`;
